### PR TITLE
Check if graphviz backend is installed

### PR DESCRIPTION
**Pull request by floscha**
Based on the discussion at the bottom of #381, this PR checks if a working Graphviz backend is installed. If not, it will suggest how to do so like described [here](https://github.com/Featuretools/featuretools/issues/381#issuecomment-459434343).

While #389 added similar hints for the documentation, I'm afraid most people will install the package without reading the docs and then face missing backend issues during runtime.

### CODE REVIEW

File: featuretools/entityset/entityset.py [hunk 1]
```diff
@@ -1113,6 +1117,24 @@ def plot(self, to_file=None):
                               ' (See https://docs.featuretools.com/getting_started/install.html for' +
                               ' details)')
 
+        # Try rendering a dummy graph to see if a working backend is installed
+        test_graph_name = 'test_graph'
+        try:
+            graphviz.Graph(test_graph_name).render()
+        except graphviz.backend.ExecutableNotFound:
+            print("To plot entity sets, a graphviz backend is required.",
```
Comment (rwedge) [inline featuretools/entityset/entityset.py]: Can this raise a RuntimeError with this message instead

File: featuretools/entityset/entityset.py [hunk 2]
```diff
@@ -1113,6 +1117,24 @@ def plot(self, to_file=None):
...
+                  "Try one the following:\n",
+                  "Mac OS: brew install graphviz\n",
+                  "Linux (Ubuntu): sudo apt-get install graphviz\n",
+                  "Windows: conda install python-graphviz",
+                  file=sys.stderr)
+            return  # Exit plotting method at this point
+        finally:
+            if os.path.isfile(test_graph_name + '.gv'):
```
Comment (rwedge) [inline featuretools/entityset/entityset.py]: Is it possible to render using a temporary file or fileobj instead

File: featuretools/entityset/entityset.py [hunk 3]
```diff
@@ -1113,6 +1117,19 @@ def plot(self, to_file=None):
...
+            graphviz.Digraph().pipe()
+        except graphviz.backend.ExecutableNotFound:
+            raise RuntimeError(
+                "To plot entity sets, a graphviz backend is required.\n" +
+                "Install the backend using one of the following commands:\n" +
+                "  Mac OS: brew install graphviz\n" +
+                "  Linux (Ubuntu): sudo apt-get install graphviz\n" +
+                "  Windows: conda install python-graphviz"
+            )
+            return  # Exit plotting method at this point
```
Comment (kmax12) [inline featuretools/entityset/entityset.py]: no need for the return after raise the RuntimeError

Comment (floscha) [reply featuretools/entityset/entityset.py]: Fair enough. Removed the *return* statement with 6df0764.

**Comment by kmax12** [comment]
Looks good. Merging

**Pull request merged**